### DISCUSSION TITLE
Is there any version can be used with jdk1.6

### DISCUSSION THREAD

**Discussion by JoshuaXX** [classgraph/classgraph#501] [General]
Hi Guys:

I am intended to use this library with jdk1.6, is this possible ?

Regards,
Joshua

**Comment by lukehutch** [comment]
I'm really sorry, no, it was a lot of work even to get it working with Java 1.7. Actually for the next major release I was considering making Java 1.8 the minimum requirement.

However as long as you're building on a JDK with version at least 1.7 (even if you're targeting 1.6), for many applications you can do all the scanning at build-time, on the build machine, and just save the scan results somewhere that the 1.6 runtime can access them:

https://github.com/classgraph/classgraph/wiki/Build-Time-Scanning

**Comment by lukehutch** [comment]
PS feel free to try to port the code to Java 1.6, if you're feeling ambitious! You could use the Animal Sniffer Maven plugin to find parts of the code that are incompatible with Java 1.6 (or just try changing the target version to 1.6, and fix the errors that come up -- although Animal Sniffer also catches semantic errors, not just syntax errors between versions).

**Reply by lukehutch** [reply]
@JoshuaXX if you do port to Java 1.6, please create a branch or fork for that, then I can direct other users that need Java 1.6 support as to where they can find it.

**Comment by GedMarc** [comment]
But... why..... 

1.8 is EOL already :( there are so many downsides to enabling backwards compatibility, or even just supporting it. 

Definitely make the min 1.8, actually could even go 11 at this point - Trying to keep jackson json version 1.8 compatible is becoming a nightmare for version 3.....

-1 for this request, -100 even

**Reply by lukehutch** [reply]
I know -- I'd even be willing to make 15 the baseline. 11 was good, and is an LTS release, but 15 is the first truly polished release since 8, in my opinion. However, classgraph has thousands of users, and many don't get to choose their JVM version :-(

**Reply by GedMarc** [reply]
Yep - what I've noticed though is this major push in security which is forcing everyone to adopt the 6 month release cycle, even if its just to do a necessary security update on a library - 
The auditors are getting their way I suppose

I just know for me, I can't keep entertaining the obsolete, it's driving me bonkers - I've just started saying no now

**Reply by lukehutch** [reply]
Yeah. The need for backwards compatibility is a major impediment to progress.

Fortunately ClassGraph is super stable now... The bug rate has dropped to almost zero. So I even if I drop JDK 1.7 in version 5.x of ClassGraph, users of 1.7 can still use 4.8.x.

**Reply by GedMarc** [reply]
Uber stable man, this thing is blowing all the other scanners away - 

I've replaced all of hibernate scanners, all javassist, the jackson scanners, and way more - one scan cycle per application on one library - You have no idea the joys of deploying micro services that startup in under a second and no message loss!

What you've done is amazing!

**Reply by lukehutch** [reply]
Awesome, and I've said this before, but thanks for stress testing the heck out of it, more than any other user that I'm aware of!
